### PR TITLE
Fix test case build errors

### PR DESCRIPTION
**Pull request by willfaught**
Some tests were failing because the test template had `e` hard-coded as the production name, but exported Go names are uppercase.

I added a `Production` template that capitalizes the param for Go and just returns the param for the other languages.

Edit: Wrong word

### CODE REVIEW

**Comment by willfaught** [comment]
Updated to fix the conflicts

**Comment by willfaught** [comment]
Please make sure to check me on these parts:

    Production(p) ::= <%<p>%>
    Production(p) ::= <%<p; format="cap">%>
    Production(p) ::= <%<p>%>
    <Production("e")>
    <currentRule.recognizerName>

Of note, I'm not sure `<Production("e")>` is going to produce `E` for Go and `e` otherwise like I want; and I'm not sure `<currentRule.recognizerName>` is going to work (do I need to declare that field as part of the model? I saw some fields declared that way, some others not, and they all seemed to be reference-able either way).

**Comment by willfaught** [comment]
Rebased

**Comment by willfaught** [comment]
Hold off on this, it's causing failures, will investigate soon.

**Comment by willfaught** [comment]
@pboyer @wjkohnen This is ready for review. I got it down to 11 failures and 1 error.

**Comment by willfaught** [comment]
Note that I removed GoTarget.supportsOverloadedMethods, so, without that override, Target.supportsOverloadedMethods for Go is now true instead of false. I did this to generate `AllFoo() []Foo` methods for multiple Foos in a rule/production and simplifying `Foo(n)` to return Foo instead of []Foo.

**Comment by willfaught** [comment]
@pboyer @wjkohnen Ping? :(

File: runtime/Go/antlr/error_strategy.go [hunk 1]
```diff
@@ -26,6 +26,8 @@ type DefaultErrorStrategy struct {
 	lastErrorStates   *IntervalSet
 }
 
+var _ ErrorStrategy = &DefaultErrorStrategy{}
```
Comment (pboyer) [inline runtime/Go/antlr/error_strategy.go:29]: @willfaught Any reason why we're creating this unused var?

File: runtime/Go/antlr/error_strategy.go [hunk 2]
```diff
@@ -737,6 +739,8 @@ type BailErrorStrategy struct {
 	*DefaultErrorStrategy
 }
 
+var _ ErrorStrategy = &BailErrorStrategy{}
```
Comment (pboyer) [inline runtime/Go/antlr/error_strategy.go]: Same here.

File: runtime/Go/antlr/error_strategy.go (hunk 1 above)
Comment (pboyer) [inline runtime/Go/antlr/error_strategy.go:29]: Actually forget it - I see you're asserting this provides an impl of ErrorStrategy!

File: runtime/Go/antlr/error_strategy.go [hunk 3]
```diff
@@ -763,8 +767,10 @@ func (b *BailErrorStrategy) Recover(recognizer Parser, e RecognitionException) {
 // Make sure we don't attempt to recover inline if the parser
 // successfully recovers, it won't panic an exception.
 //
-func (b *BailErrorStrategy) RecoverInline(recognizer Parser) {
+func (b *BailErrorStrategy) RecoverInline(recognizer Parser) Token {
```
Comment (pboyer) [inline runtime/Go/antlr/error_strategy.go:770]: Nice catch!

**Comment by pboyer** [comment]
My god I was slow on this. I owe you one. This is great - thank you so much @willfaught!

File: runtime/Go/antlr/error_strategy.go (hunk 1 above)
Comment (willfaught) [inline runtime/Go/antlr/error_strategy.go:29]: Yep! :)

**Comment by willfaught** [comment]
Thanks for merging!

**Pull request merged**